### PR TITLE
Bump mailer from 1.33 to 1.34

### PR DESCRIPTION
**Pull request by basil**
Bumps [Mailer](https://plugins.jenkins.io/mailer/) to the latest version and removes a now-unneeded entry from the `<dependencyManagement>` section that was originally added to deal with a `requireUpperBoundDeps` error that is no longer present. May or may not require a revert of commit cbe2b0f1d3 to adapt to jenkinsci/mailer-plugin#112 but I can't run the test locally to find out, so let's see what happens in the CI run.

### CODE REVIEW

**Comment by basil** [comment]
The tests pass _with_ commit cbe2b0f; now let's see what happens without it…

**Comment by basil** [comment]
Reverting commit cbe2b0f failed with

```
org.junit.ComparisonFailure: expected:<...yleProject":0,"org-j[]vnet-hudson-test-Moc...> but was:<...yleProject":0,"org-j[enkinsci-plugins-workflow-job-WorkflowJob":0,"org-j]vnet-hudson-test-Moc...>
	at org.junit.Assert.assertEquals(Assert.java:117)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at hudson.model.UsageStatisticsTest.compareWithFile(UsageStatisticsTest.java:172)
	at hudson.model.UsageStatisticsTest.roundtrip(UsageStatisticsTest.java:121)
```

so I guess that commit is needed after all. So I undid the revert. This should be ready to review now.

**Review by res0nance — APPROVED** [review]

File: test/pom.xml
Comment (oleg-nenashev) [inline test/pom.xml:59]: Maybe it makes sense to keep it so that we do not run into the issues in the future

Comment (basil) [reply test/pom.xml:59]: Maybe, but are we sure that the current `<dependencyManagement>` section will always apply to future `requireUpperBoundDeps` errors? It may or may not, so I think it is best to re-evaluate on a case by case basis. I think the ideal permanent solution would be to start using the plugin BOM, and/or remove plugin dependencies from the Jenkins core test suite.

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
